src/scala/co/actioniq/luna/dao/IgnoreUpdateCompiler.scala

```diff
@@ -0,0 +1,111 @@
+package co.actioniq.luna.dao
+
+import slick.compiler.InsertCompiler.Mode
+import slick.compiler.{CompilerState, Phase}
+import slick.util.ConstArray
+
+/**
+  * The goal of this compile stage is to remove fields from update statements that should not be updated when using
+  * case classes.  This is useful for updating using a case class and ignoring indexed fields, which would cause a
+  * deadlock.  This only removes fields from updating using a case class, sqlu and updating specific fields
+  * are not affected by this compiler.
+  *
+  * How it works:
+  * 1. Assumes ResultSetMapping state (probably after hoistClientOps).
+  * 2. Looks for fields in "from" with column option of IgnoreFieldForUpdate.  Removes those.
+  * 3. Removes from mapper fields with same ordinal position as fields in #2.
+  * 4. Creates a new "wrapper" function around mapper (mapper function goes from case class to tuple)
+  * that will create a tuple without ignored fields.
+  */
+class IgnoreUpdateCompiler extends Phase {
+  import slick.ast._
+  override val name: String = "ignoreUpdateCompiler"
+
+  override def apply(state: CompilerState): CompilerState = {
+    var fieldsToRemove: Map[Int, Boolean] = Map()
+    state.map {
+      case rsm @ ResultSetMapping(_, from, map) =>
+        val newFrom = from match {
+          case b @ Bind(sym, _, bindSelect) =>
+            val newSelect = bindSelect match {
+              case p @ Pure(pureSelect, _) =>
+                val filteredSelect = pureSelect match {
+                  case struct @ StructNode(elements) if elements.forall{ case (_, Select(Ref(str), _)) if str == sym => true; case _ => false} =>
```

Comment:
can you pull this if-condition into a well-named function? I have no idea what it's trying to check for right now

Reply:
i copied this from slick code lol but i can make it better